docs/actions/transformer_actions/README.md

```diff
@@ -0,0 +1,772 @@
+
+The resulting template after these edits are:
+```python
+from mage_ai.data_cleaner.transformer_actions.base import BaseAction
+from mage_ai.data_cleaner.transformer_actions.constants import ActionType, Axis
+from mage_ai.data_cleaner.transformer_actions.utils import build_transformer_action
+from pandas import DataFrame
+
+if 'transformer' not in globals():
+    from mage_ai.data_preparation.decorators import transformer
+
+
+@transformer
+def execute_transformer_action(df: DataFrame, *args, **kwargs) -> DataFrame:
+    """
+    Execute Transformer Action: ActionType.AVERAGE
+    """
+    action = build_transformer_action(
+        df,
+        action_type=ActionType.AVERAGE,
+        action_code='',  # Enter filtering condition on rows before aggregation
+        arguments=['Fare', 'Age'],  # Enter the columns to compute aggregate over
+        axis=Axis.COLUMN,
+        options={'groupby_columns': ['Embarked']},  # Enter columns to group by
+        outputs=[
+            # The number of outputs below must match the number of arguments
+            {'uuid': 'avg_fare_by_embarked_loc', 'column_type': 'number_with_decimals'},
+            {'uuid': 'avg_age_by_embarked_loc', 'column_type': 'number_with_decimals'},
+        ],
+    )
+
+    return BaseAction(action).execute(df)
+```
+Run this transformer, and your output data frame should now include two new columns named "avg_fare_by_embarked_loc" and "avg_age_by_embarked_loc" which contain the average fare and average age by embarkation location. Each row gets a new value corresponding to the group which that row falls into.
+
+This workflow can be summarized to describe how Transformer Actions can be used effectively:
+1. Identify a transformation that may need to be made to your data
+2. Create a transformer using the transformer action that corresponds to your template
+3. Fill in the parameters for the transformer action to build your transformer action payload
+4. Build a `BaseAction` object using the payload and execute the action to get the transformed data frame
+
+
+## Core Concepts
+
+The core object of any transformer action is the **payload**, a JSON object that describes the transformation to perform alongside settings for that transformation. A sample transformer action payload is shown below:
+
+```json
+{
+    "action_type": "impute",
+    "action_arguments": ["age", "name"],
+    "action_options": {
+        "strategy": "constant",
+        "value": "32"
+    },
+    "axis": "column",
+    "action_variables": {
+        "age": {
+            "uuid": "age",
+            "column_type": "number"
+        },
+        "name": {
+            "uuid": "name",
+            "column_type": "text"
+        }
+    }
+}
+```
+The above transformer action imputes missing values in the 'age' and "name" columns with the constant value `32`. Here is a list of the keys used in transformer action payloads:
```

Comment:
nit: age uses single quote and name uses double quotes. let's make it consistent here. also filling in name with a number 32 might be confusing.

Reply:
I changed "name" to "hourly_salary" so a number makes sense, and both now use double quotes.